pkg/cli/info.go

```diff
@@ -72,3 +93,30 @@ func (s *Info) Run(cmd *cobra.Command, args []string) error {
 	})
 	return out.Err()
 }
+
+func removeUnsetFields(spec v1.InfoSpec, configKeys ...string) (map[string]any, error) {
+	b, err := json.Marshal(spec)
+	if err != nil {
+		return nil, err
+	}
+
+	var specMap map[string]any
+	if err := json.Unmarshal(b, &specMap); err != nil {
+		return nil, err
+	}
+
+	for _, key := range configKeys {
+		if cfg, ok := specMap[key].(map[string]any); ok {
+			for _, entry := range typed.Sorted(cfg) {
+				if entry.Value == nil || reflect.ValueOf(entry.Value).IsZero() {
+					delete(cfg, entry.Key)
+				}
+			}
+			if len(cfg) == 0 {
+				delete(specMap, key)
+			}
+		}
+	}
+
+	return specMap, nil
```

Comment:
I _think_ there's a more concise way to do this involving [gojq](https://pkg.go.dev/github.com/itchyny/gojq#Parse) and a query like `del(.. | select(. == null or . == ""))`.

Reply:
I'm not familiar with gojq, so your suggestion doesn't seem as understandable or concise to me. I would also be curious whether using gojq would be less performant than using built-in Go functionality.

Do you think it would be worth it to explore the performance trade offs?

Reply:
> I'm not familiar with gojq, so your suggestion doesn't seem as understandable or concise to me.

TLDR; Please ignore this suggestion! Sorry!

I used it to solve a similar problem in the past, but it turned out the example I was thinking of wasn't as clean as I remembered. Meant to close this out earlier.